core/nixpkgs.bzl

```diff
@@ -106,18 +106,40 @@ def nixpkgs_git_repository(
     )
 
 def _nixpkgs_local_repository_impl(repository_ctx):
-    if not bool(repository_ctx.attr.nix_file) != \
-       bool(repository_ctx.attr.nix_file_content):
-        fail("Specify one of 'nix_file' or 'nix_file_content' (but not both).")
+    if bool(repository_ctx.attr.nix_file) and bool(repository_ctx.attr.nix_file_content) or \
+       bool(repository_ctx.attr.nix_file) and bool(repository_ctx.attr.nix_flake_lock_file) or \
+       bool(repository_ctx.attr.nix_flake_lock_file) and bool(repository_ctx.attr.nix_file_content):
+        fail("Specify only one of 'nix_file', 'nix_file_content' or 'nix_flake_lock_file'.")
+
     if repository_ctx.attr.nix_file_content:
         target = "default.nix"
         repository_ctx.file(
             target,
             content = repository_ctx.attr.nix_file_content,
             executable = False,
         )
-    else:
+    elif repository_ctx.attr.nix_file:
         target = cp(repository_ctx, repository_ctx.attr.nix_file)
+    else:
```

Comment:
Minor, but it looks like the `if` statement above doesn't catch the case where all of `nix_file`, `nix_file_content` and `nix_flake_lock_file` are `None`? Maybe this `else` could be `elif repository_ctx.attr.nix_flake_lock_file`, with an `else` at the end to `fail` when none of them are set.